components/HomeHeader.vue

```diff
@@ -124,50 +131,69 @@ export default Vue.extend({
     this.items = (
       await this.$api.userLibrary.getLatestMedia({
         userId: this.$auth.user.Id,
-        limit: 10,
+        limit: this.pages,
         fields: [ItemFields.Overview],
         enableImageTypes: [ImageType.Backdrop, ImageType.Logo],
         imageTypeLimit: 1
       })
     ).data;
-  },
-  methods: {
-    getBackdrop(item: BaseItemDto): string {
-      if (item.Type === 'Episode') {
-        return this.getImageUrlForElement(ImageType.Backdrop, {
-          itemId: item.SeriesId
-        });
-      } else if (item.Type === 'MusicAlbum') {
-        return this.getImageUrlForElement(ImageType.Backdrop, {
-          itemId: item.AlbumArtists?.[0].Id
-        });
-      } else {
-        return this.getImageUrlForElement(ImageType.Backdrop, {
-          itemId: item.Id
-        });
-      }
-    },
-    getLogo(item: BaseItemDto): string {
-      if (item.Type === 'Episode' && item.SeriesId) {
-        return this.getImageUrlForElement(ImageType.Logo, {
-          itemId: item.SeriesId
-        });
-      } else if (item.ParentLogoItemId) {
-        return this.getImageUrlForElement(ImageType.Logo, {
-          itemId: item.ParentLogoItemId as string
-        });
+
+    for (const [key, i] of this.items.entries()) {
+      let id: string;
+      if (i.Type === 'Episode' && i.SeriesId) {
+        id = i?.SeriesId as string;
+      } else if (i.Type === 'MusicAlbum') {
+        id = i?.AlbumArtists?.[0].Id as string;
+      } else if (i.ParentLogoItemId) {
+        id = i?.ParentLogoItemId as string;
       } else {
-        return this.getImageUrlForElement(ImageType.Logo, {
-          item
-        });
+        continue;
       }
+
+      const itemData = (
+        await this.$api.userLibrary.getItem({
+          userId: this.$auth.user.Id,
+          itemId: id
+        })
+      ).data;
```

Comment:
We should need this, since we should have everything required in what's returned by the latest endpoint (If we don't, it's better to add the info there (for things like blurhashes and such), imo.

Reply:
@mrtimscampi We don't: what we do here is to get parent items.

This function runs the exact same checks you added for getting the image backdrops: for music, it fetched artist, for series, it fetched series item.

As we now use BlurhashImage, we need to pass the full item object. That's where this function enters the game: it requests the full item object of the items that use the backdrop from the parent.

I store those "Related Items" in a dictionary (key is the index of the element in the this.items array) and value this related item.

Reply:
> As we now use BlurhashImage, we need to pass the full item object. That's where this function enters the game: it requests the full item object of the items that use the backdrop from the parent.

I still think we should change the server to include the blurhashes from the parent items as well.

We already get the image tags and item IDs for everything, so it'd make sense to get that as well, to minimize requests.

However, for now, it's necessary. Please add a TODO or HACK mentioning we need to go back to this when the server changes have been made, though.